src/storage/src/metrics/source.rs

```diff
@@ -86,42 +80,40 @@ impl GeneralSourceMetricDefs {
             progress: registry.register(metric!(
                 name: "mz_source_progress",
                 help: "A timestamp gauge representing forward progess in the data shard",
-                var_labels: ["source_id", "output", "shard", "worker_id"],
+                var_labels: ["source_id", "shard", "worker_id"],
```

Comment:
I removed all notions of "output index" from metrics. The index is not exposed anywhere externally and even in our logs we identify everything by its id.